pkg/option/config.go

```diff
@@ -3055,6 +3055,12 @@ func (c *DaemonConfig) calculateDynamicBPFMapSizes(totalMemory uint64, dynamicSi
 			getEntries(NATMapEntriesGlobalDefault, LimitTableAutoNatGlobalMin, LimitTableMax)
 		log.Infof("option %s set by dynamic sizing to %v",
 			NATMapEntriesGlobalName, c.NATMapEntriesGlobal)
+		if c.NATMapEntriesGlobal > c.CTMapEntriesGlobalTCP+c.CTMapEntriesGlobalAny {
+			// CT table size was specified manually, make sure that the NAT table size
+			// does not exceed maximum CT table size. See
+			// (*DaemonConfig).checkMapSizeLimits.
+			c.NATMapEntriesGlobal = int((c.CTMapEntriesGlobalTCP + c.CTMapEntriesGlobalAny) * 2 / 3)
```

Comment:
So in case dynamic sizing is enabled but map sizes are also explicitly specific, we will follow the dynamic sizing up to the explicit size? Do we have a note on that behavior in the documentation already? I think we should also print an info or warning message in logs since several Cilium flags are at odds.

Reply:
I don't think we have a note on that behavior in our docs yet, will add one (as well as a warning message for this case).